src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid.tsx

```diff
@@ -97,6 +97,26 @@ export const MetricsGrid = ({
     setExpandedMetric(undefined);
   }, []);
 
+  // TODO: find a better way to handle conflicts with other flyouts
+  // Close the metric insights flyout if the user clicks on the inspector button
+  useEffect(() => {
+    const handleClick = (event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+
+      if (target.closest('[data-test-subj="embeddablePanelAction-openInspector"]')) {
+        if (expandedMetric) {
+          handleCloseFlyout();
```

Comment:
Q: As we have the `DiscoverFlyouts.inspectorPanel` would the `handleViewDetails` already close it? There we call `dismissAllFlyoutsExceptFor(DiscoverFlyouts.metricInsights);`

Reply:
Ah, I see, thanks for explaining. I would say let's investigate further and add an issue to improve that and see if there is an option to enhance the lens component to accept extra on click actions instead in a separate issue. Wdyt?